src/states/node.rs

```diff
@@ -1654,6 +1736,64 @@ impl Node {
         true
     }
 
+    fn send_own_group_merge(&mut self,
+                            targets: Vec<XorName>,
+                            merge_details: OwnMergeDetails<XorName>,
+                            src: Authority) {
+        for target in &targets {
+            let (connected, needed): (Vec<_>, Vec<_>) = merge_details.groups
+                .clone()
+                .into_iter()
+                .partition(|&(prefix, _)| merge_details.prefix.is_compatible(&prefix));
```

Comment:
I think for that optimisation we should rather check whether `<target's prefix>.is_neighbour(&prefix)` - in that case, it will already be connected.

I don't think the merging group already has all connections. In my example above, `00` won't be connected to `11`, but `0100` is already connected to that, and the new group `0` will need to be.